### ISSUE TITLE
prompt from external JS file not working

### ISSUE THREAD

**Issue by hakiemaul-flai** [promptfoo/promptfoo#1720]
**Describe the bug**
When using prompt from external file
```
prompts:
  - file://prompt.js

providers:
  - label: "fun"
    id: "openai:gpt-4o-mini"

tests:
  - vars:
      topic: bananas
```

with `prompt.js` written like
```
module.exports = async function ({ vars }) {
    return `Write a tweet about ${vars.topic}`;
};
```

The input prompt into openAI provider seems to be an actual code
```
[
  {
    role: 'user',
    content: 'async function ({ vars }) {\n' +
      '    return `Write a tweet about ${vars.prompt}`;\n' +
      '  }'
  }
]
```

**To Reproduce**
Install latest version of promptfoo, define a prompt as a JS file, and try running.

**Expected behavior**
JS file to run and resolved to be a string as returned

**Screenshots**

**System information (please complete the following information):**

- Promptfoo version:
0.89.0

**Additional context**
Previously working fine on version 0.87.1
might be similar issue as #1717 and #1716 as well

**Comment by albertlieyingadrian** [comment]
Seems like it breaks after this release https://github.com/promptfoo/promptfoo/compare/0.87.1...0.88.0

**Comment by albertlieyingadrian** [comment]
The exact line that is problematic https://github.com/promptfoo/promptfoo/issues/1716#issuecomment-2362639691

**Comment by albertlieyingadrian** [comment]
@mldangelo I created a PR to handle the case when it's a javascript function in case you want to merge your PR back

https://github.com/promptfoo/promptfoo/pull/1724

**Comment by hakiemaul-flai** [comment]
fixed with #1724 , thanks @albertlieyingadrian
